__tests__/tasks/profile-picture.test.js

```diff
@@ -0,0 +1,90 @@
+const puppeteer = require('puppeteer');
+const API_BASE_URL = 'https://staging-api.realdevsquad.com';
+const { members } = require('../../mock-data/members');
+
+describe('Task Page - Assignee Profile Pic', () => {
+  let browser;
+  let page;
+  jest.setTimeout(60000);
+
+  beforeAll(async () => {
+    browser = await puppeteer.launch({
+      headless: true,
+      ignoreHTTPSErrors: true,
+      args: ['--incognito', '--disable-web-security'],
+      devtools: false,
+    });
+    page = await browser.newPage();
+    await page.setRequestInterception(true);
+
+    page.on('request', (interceptedRequest) => {
+      const url = interceptedRequest.url();
+
+      if (url === `${API_BASE_URL}/members`) {
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(members),
+        });
+      } else {
+        interceptedRequest.continue();
+      }
+    });
+    await page.goto('http://localhost:8000/task/');
+    await page.waitForNetworkIdle();
+
+    // Click the button and select the status as Assigned
+    const buttonId = 'statusId';
+    const button = await page.$(`#${buttonId}`);
+    await button.click();
+    await page.waitForSelector('select.input:not(.notEditing)');
+    await page.select('select#status', 'ASSIGNED');
+  });
+
+  afterAll(async () => {
+    await browser.close();
+  });
+
+  it('Profile picture with url', async () => {
+    const inputElement = await page.$('input#assignee');
+    const name = 'satyam';
+
+    //Trigger the input event manually
+    for (let i = 0; i < name.length; i++) {
+      await inputElement.type(name[i], { delay: 100 });
+    }
+
+    await page.waitForSelector('#suggested-users-container');
+
+    await page.waitForTimeout(2000);
+    const imgSrc = await page.$eval('#list-items img', (img) => img.src);
+    const expectedImageFilename =
+      'https://res.cloudinary.com/imgprc/image/upload/v1688145505/profile/tu4zVmwFX91e9oEuSWLg/woqilcqdbp0cudcexkem.jpg';
+    expect(imgSrc.endsWith(expectedImageFilename)).toBe(true);
+  });
```

Comment:
How are we getting the actual URL here?
Are we making actual API call and checking?
Tests should not be doing actual network calls.

Reply:
No, I have created mock data and in that mocked data there is url and when the api is getting hit I  am mocking the api response and sending that mock data and checking the value of url present in mock data equal to src element of img tag.
![image](https://github.com/Real-Dev-Squad/website-dashboard/assets/37043221/eec0eab9-5459-4803-bbb4-86e869404850)